iceoryx_hoofs/platform/win/source/handle_translator.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/platform/handle_translator.hpp"
+
+HandleTranslator& HandleTranslator::getInstance() noexcept
+{
+    static HandleTranslator globalHandleTranslator;
+    return globalHandleTranslator;
+}
+
+HANDLE HandleTranslator::get(const int handle) const noexcept
+{
+    return m_handleList[static_cast<size_t>(handle)];
+}
+
+int HandleTranslator::add(HANDLE handle) noexcept
+{
+    for (int64_t limit = m_handleList.size(), k = 0; k < limit; ++k)
```

Comment:
It's a pity we don't get the index in a range based for loop. But why this unusual for-loop parameters and not just `int64_t i = 0; i < m_handleList.size(); ++i`

Reply:
It is my habit and this is useful when working with a `std::vector` where it is possible that with every iteration the size is determined again. But I admit this is not required in the `cxx::vector` case since we store the size there.

Reply:
It's the same with `std::vector` and other container, at least in release builds.